aether-ui/aether/ui/api/tests/test_views.py

```diff
@@ -170,65 +182,75 @@ def test_view_pipeline___publish(self):
         })
         response = self.client.post(url, data=data, content_type='application/json')
         response_data = json.loads(response.content)
-        pipeline_id = response_data['id']
-        pipeline2 = Pipeline.objects.get(pk=pipeline_id)
+        contract_id = response_data['id']
+        contract3 = Contract.objects.get(pk=contract_id)
+        contract3.kernel_refs = contract.kernel_refs
+        contract3.save()
         url = reverse('pipeline-publish', args=[pipeline_id])
-        response = self.client.post(url, {'project_name': 'Aux 1'})
+        response = self.client.post(url, {'project_name': 'Aux 1', 'contract_id': contract_id})
         response_data = json.loads(response.content)
         self.assertGreater(len(response_data['exists']), 0)
+        self.assertIn(response_data['exists'][0]['Screening'],
+                      '{} schema with id {} exists'.format('Screening',
+                                                           contract3.kernel_refs['schemas']['Screening']))
 
         url = reverse('pipeline-publish', args=[str(pipeline_id) + 'wrong'])
         response = self.client.post(url, {'project_name': 'Aux 1'})
         response_data = json.loads(response.content)
-
-        pipeline.kernel_refs = {}
-        pipeline.save()
-
+        self.assertIn('is not a valid UUID', response_data['error'][0])
         outcome = {
                     'successful': [],
                     'error': [],
-                    'exists': []
+                    'exists': [],
+                    'ids': {
+                        'mapping': {},
+                        'schema': {},
+                    }
                 }
-        outcome = utils.publish_preflight(pipeline, 'Aux', outcome)
-        self.assertEqual(len(outcome['exists']), 5)
+        outcome = utils.publish_preflight(pipeline, 'Aux', outcome, contract)
+        self.assertEqual(len(outcome['exists']), 3)
 
     def test_view_pipeline__publish(self):
         pipeline = Pipeline.objects.create(
             name='Pipeline Mock 2',
             schema={'name': 'test', 'type': 'record', 'fields':
                     [{'name': 'name', 'type': 'string'}]},
             input={'test': {'name': 'myValue'}},
+        )
+        contract = Contract.objects.create(
             entity_types=[{'name': 'Test', 'type': 'record', 'fields':
```

Comment:
Styling question, what do you find more readable?

1.
```
[{'name': 'Test', 'type': 'record', 'fields':
 [{'type': 'string', 'name': 'name'}, {'type': 'string', 'name': 'id'}]}],
```

2.
```
[{
  'name': 'Test', 
  'type': 'record', 
  'fields': [{'type': 'string', 'name': 'name'}, {'type': 'string', 'name': 'id'}]
}],
```

3.
```
[{
  'name': 'Test', 
  'type': 'record', 
  'fields': [
    {'type': 'string', 'name': 'name'},
    {'type': 'string', 'name': 'id'}
  ]
}],
```

Reply:
🤣